### PR TITLE
Add license enforcement

### PR DESCRIPTION
**Pull request by jnummelin**
Running OSS version outputs:
```
==> Configure license enforcement @ demo-pharos-master-0
    [demo-pharos-master-0] No license enforcement on Pharos OSS ...
    [demo-pharos-master-0] Completed phase in 0.00s
```

Running non-oss version deploys the actual license enforcer into the cluster.

Need to still modify the image reference once multiarch images are built.

### CODE REVIEW

File: lib/pharos/cluster_manager.rb [hunk 1]
```diff
@@ -101,6 +101,7 @@ def apply_phases
 
       # master is now configured and can be used
       # configure essential services
+      apply_phase(Phases::ConfigureLicenseEnforcer, master_only)
```
Comment (jakolehm) [inline lib/pharos/cluster_manager.rb]: Could phases be injected from pro/ee versions? /cc @kke

Comment (kke) [reply lib/pharos/cluster_manager.rb]: Well, we could trick apply_phase:

```ruby
class ClusterManager
  def apply_phase(klass, hosts, **opts)
    super
    if klass == Phases::FooFoo
      super(Phases::ConfigureLicenseEnforcer, [config.master_host])
    end
  end
end
```

Comment (jnummelin) [reply lib/pharos/cluster_manager.rb]: we decided to go with `phase.enabled?` way...

File: lib/pharos/phase_manager.rb [hunk 2]
```diff
@@ -25,6 +26,7 @@ def self.load_phases(*dirs)
     # @param dirs [Array<String>]
     def initialize(**options)
       @options = options
+      @pastel = options.delete(:pastel)
```
Comment (kke) [inline lib/pharos/phase_manager.rb]: Not needed

File: lib/pharos/phases/configure_license_enforcer.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require_relative 'mixins/psp'
```
Comment (jakolehm) [inline lib/pharos/phases/configure_license_enforcer.rb]: Leftovers?

File: non-oss/pharos_pro/phases/configure_license_enforcer.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+module Pharos
+  module Phases
+    class ConfigureLicenseEnforcer < Pharos::Phase
+      title "Configure license enforcement"
+
+      LICENSE_ENFORCER_VERSION = '0.1.0'
+      RESOURCE_PATH = Pathname.new(File.expand_path(File.join(__dir__, '..', 'resources', 'license-enforcer'))).freeze
+
+      register_component(
+        name: 'PHAROS-LICENSE-ENFORCER', version: LICENSE_ENFORCER_VERSION, license: 'Kontena License'
```
Comment (jakolehm) [inline non-oss/pharos_pro/phases/configure_license_enforcer.rb]:
```suggestion
        name: 'Pharos License Enforcer', version: LICENSE_ENFORCER_VERSION, license: 'Kontena License'
```

File: lib/pharos/phases/configure_license_enforcer.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  module Phases
+    class ConfigureLicenseEnforcer < Pharos::Phase
+      title "Configure license enforcement"
+
+      def enabled?
+        false
+      end
+
+      def call
+        logger.info { "No license enforcement on Pharos OSS ..." }
```
Comment (jakolehm) [inline lib/pharos/phases/configure_license_enforcer.rb]: Not needed?

File: lib/pharos/phase_manager.rb (hunk 2 above)
Comment (jnummelin) [reply lib/pharos/phase_manager.rb]: removed

File: lib/pharos/phases/configure_license_enforcer.rb (hunk 3 above)
Comment (jnummelin) [reply lib/pharos/phases/configure_license_enforcer.rb]: removed

**Pull request closed without merging**